marge/gitlab.py

```diff
@@ -15,7 +15,7 @@ def call(self, command, sudo=None):
         url = self._api_base_url + command.endpoint
         headers = {'PRIVATE-TOKEN': self._auth_token}
         if sudo:
-            headers['SUDO'] = '%d' % sudo
+            headers['SUDO'] = f'{sudo}'
```

Comment:
This will probably be a (breaking?) change, because:

```
$ python3
Python 3.10.8 (main, Nov  4 2022, 09:21:25) [GCC 12.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> '%d' % true
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'true' is not defined. Did you mean: 'True'?
>>> '%d' % True
'1'
>>> f'{True}'
'True'
```

This could be a workaround (tested on my local Python 3.10.8):

```suggestion
            headers['SUDO'] = f'{1 if sudo else 0}'
```

Reply:
Very good catch, thanks :bow:

Reply:
Ah, but no, it looks like `f'{sudo}` actually works. It's not a boolean flag, but a user ID used like this:

```python
for uid in self.approver_ids:
    self._api.call(POST(approve_url), sudo=uid)
```

So it should actuallly output it as is if it's set.